modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue

```diff
@@ -561,6 +562,26 @@ export default {
         }
         return window.apos.modules[`${item.type}-widget`];
       });
+    },
+    // Return a new widget object in which defaults are fully populated,
+    // especially valid sub-area objects, so that nested edits work on the page
+    newWidget(type) {
+      const widget = {
+        type
+      };
+      const schema = apos.modules[apos.area.widgetManagers[type]].schema;
+      schema.forEach(field => {
+        if (field.type === 'area') {
+          widget[field.name] = {
+            _id: cuid(),
```

Comment:
Without this, the server side has to invent its own id for the area, which doesn't match the one the browser winds up inventing. Make sure we send the area up with a proper shape and a proper id in the first place when we add a widget without an initial modal.

In the presence of the editor modal this is not a problem because the editor winds up generating an id and that gets sent to the server the first time and used consistently thereafter.

Reply:
I see, well spotted!